### PR TITLE
Messages bugfix again :D

### PR DESCRIPTION
**Pull request by jfrader**
Handle offline messages 
Fetch messages and groups when broadcasted

### CODE REVIEW

**Comment by RockySouthpaw** [comment]
**Fixed**
- Offline Messages
 These no longer toss any errors and prints the player is offline correctly.
- Pressing `enter` resulting in the group being created.


This pr doesn't seem to fix the issue with creating a group that already exist as it still loads infinitely. 

Still need to test the following and will edit this after:
- Message notifications loading infinitely when you click on them.
- Message fetching.

**Comment by jfrader** [comment]
> This pr doesn't seem to fix the issue with creating a group that already exist as it still loads infinitely.
> 
Can you test this again? Pushed a possible fix.

**Comment by RockySouthpaw** [comment]
Testing now

**Comment by RockySouthpaw** [comment]
> > This pr doesn't seem to fix the issue with creating a group that already exist as it still loads infinitely.
> 
> Can you test this again? Pushed a possible fix.

Retested. Issue still exist.

**Comment by RockySouthpaw** [comment]
image

Not sure if this was the result of this branch as I currently can't retest it. But this was a result of a notification being triggered with the phone closed.

**Comment by jfrader** [comment]
No, this is probably new ESX removed branch.

**Comment by RockySouthpaw** [comment]
This PR is already getting too big. 

> No, this is probably new ESX removed branch.

This was prior to this branch being re-based. Will re test later.

**Comment by RockySouthpaw** [comment]
Message groups don't work at all now.

**Comment by RockySouthpaw** [comment]
Looks like this commit is the issue: https://github.com/project-error/new-phone-who-dis/pull/164/commits/ce892c6f5dfb095c41e6b513a6dd59b40537f44c Reverting it fixes

File: phone/src/apps/messages/utils/goToConversation.ts [hunk 1]
```diff
@@ -1,2 +1,7 @@
-export const goToConversation = (messageGroup, history) =>
+import Nui from '../../../os/nui-events/utils/Nui';
+
+export const goToConversation = (messageGroup, history) => {
+  if (!messageGroup.groupid || !history) return;
```
Comment (itschip) [inline phone/src/apps/messages/utils/goToConversation.ts:4]: Seems to be a typo in 'groupid'. Should it not be 'groupId'?

**Review by itschip — CHANGES_REQUESTED** [review]

**Comment by jfrader** [comment]
Yes, that is the bug

**Review by RockySouthpaw — APPROVED** [review]
Tested and fixes the major issues. Some small ones left to task to simple pr's but the right direction so.

**Pull request merged**